slicerator/__init__.py

```diff
@@ -361,25 +362,37 @@ def _index_generator(new_indices, old_indices):
 
 
 class Pipeline(object):
-    def __init__(self, ancestor, proc_func, propagate_attrs=None):
+    def __init__(self, proc_func, *ancestors, **kwargs):
         """A class to support lazy function evaluation on an iterable.
 
         When a ``Pipeline`` object is indexed, it returns an element of its
         ancestor modified with a process function.
 
         Parameters
         ----------
-        ancestor : object
         proc_func : function
             function that processes data returned by Slicerator. The function
             acts element-wise and is only evaluated when data is actually
             returned
+        *ancestors : objects
+            Object to be processed.
+        propagate_attrs : set of str or None, optional
+            Names of attributes to be propagated through the pipeline. If this
+            is `None`, go through ancestors and look at `_propagate_attrs`
+            and `propagate_attrs` attributes and search for attributes having
+            a `_propagate_flag` attribute. Defaults to `None`.
+        propagate_how : {'first', 'last'} or int, optional
+            Where to look for attributes to propagate. If this is an integer,
+            it specifies the index of the ancestor (in `ancestors`). If it is
+            'first', go through all ancestors starting with the first one until
+            one is found that has the attribute. If it is 'last', go through
+            the ancestors in reverse order. Defaults to 0.
```

Comment:
I would prefer to stick to `int` here and using negative integers for indexing from the end of the `ancestors` list. So `0` equals `'first'` and `-1` equals `'last'`. Probably the implementation will simplify because this is normal Python list indexing.

Reply:
Sorry I now understand the necessity of this implementation, please ignore my comment.